ci_framework/roles/os_must_gather/tasks/main.yml

```diff
@@ -21,6 +21,21 @@
   loop:
     - logs
 
+- name: Construct project change list
+  ansible.builtin.set_fact:
+    zuul_change_list: "{{ zuul_change_list | default([]) + [item.project.short_name] }}"
+    cacheable: true
+  with_items: "{{ zuul['items'] }}"
+  when:
+    - zuul is defined
+    - "'change_url' in item"
+
+- name: Build openstack-must-gather image
+  ansible.builtin.import_tasks: build_openstack-must-gather_image.yml
+  when:
+    - zuul_change_list is defined
+    - "'openstack-must-gather' in zuul_change_list"
```

Comment:
It is no longer needed, if we use the zuul provided repo path. "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/openstack-must-gather". May be we can replace this zuul piece with a new var to `cifmw_os_must_gather_image_build`.

Reply:
Sorry for the noise here. I withdraw my comment. If we always builds the image and push it to registry. Then it is going to be a waste of resources. It make sense to build it only on zuul changes.